sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -382,8 +382,14 @@ case class UnwrapOption(
 
   override def inputTypes: Seq[AbstractDataType] = ObjectType :: Nil
 
-  override def eval(input: InternalRow): Any =
-    throw new UnsupportedOperationException("Only code-generated evaluation is supported")
+  override def eval(input: InternalRow): Any = {
+    val inputObject = child.eval(input)
```

Comment:
NIT: you can also use pattern matching here. That saves you some typing:
```scala
child.eval(input) match {
  case Some(value) => value
  case _ => null
}
```

Reply:
thanks for your comment. I haven't used pattern matching because it is a bit slower than using if...else and I'd prefer the fastest way here. What do you think?

Reply:
I am fine either way. This is going to be slow anyway. Let's leave as it is.